### DISCUSSION TITLE
ImporterMeshInstance3D metadata lost in glTF import process

### DISCUSSION THREAD

**Discussion by emperorofmars** [godotengine/godot-proposals#6586] [3D]
Based on Godot 4.0.1

In the glTF import proces with 'GLTFDocumentExtension' i would like to set metadata on nodes.
This works fine with most node types. But for 'ImporterMeshInstance3D' all metadata gets lost.
The 'ImporterMeshInstance3D' node gets converted into 'MeshInstance3D' sometime after the 'GLTFDocumentExtension' functions have completed, so i cannot do anything about that.
I would like to have a way to preserve Metadata through the conversion from 'ImporterMeshInstance3D' to 'MeshInstance3D' and similar nodetypes if applicable.

**Comment by tamask** [comment]
I've created a pull request to resolve this issue: https://github.com/godotengine/godot/pull/87584.

**Comment by akien-mga** [comment]
Fixed by https://github.com/godotengine/godot/pull/87584.
